README.md

```diff
@@ -1,31 +1,72 @@
+
+First and foremost, it's important to note that not every device (microcontroller, module, board) is capable of running device management features. And while some hardware capabilities can be easily added or extended - like connectivity, storage, TRNG - others are impossible or inconvenient to extend (e.g. RAM/Flash).
+
+Here's a list of minimum of requirements to add device management feature to your application:
+* RAM: 128K or more
+* Flash: 512K or more
+* Real Time Clock (RTC)
+* (optional but recommended) True Random Number Generator (TRNG)
+* A storage device - SDcard, SPI Flash, QSPI Flash, Data Flash
+* IP connectivity - Ethernet, Wi-Fi, Cellular, 6LoWPAN, Thread
 
-This library is a simpler interface to Pelion DM Client, making it trivial to expose sensors, actuators and other variables to the cloud. For a complete Pelion DM CLient API, check our [documentation](https://cloud.mbed.com/docs/current/mbed-cloud-client/index.html).
+Additionally, the device and any additional complementary hardware components should be supported and provided in the latest releases of Mbed OS, or have support for the APIs provided in the latest releases of Mbed OS.
 
-## Usage to Connect to Pelion Device Management
+#### Useful references
+
+* Check which Mbed OS platforms are supported in the [Pelion Device Management quick-start guide](https://cloud.mbed.com/quick-start).
+* Check which storage options are available [here](https://os.mbed.com/docs/latest/reference/storage.html).
+* Check which network options are available [here](https://os.mbed.com/docs/latest/reference/network-socket.html).
+
+
+### Adding device management feature to your application
 
 1. Add this library to your Mbed OS project:
 
     ```
     $ mbed add https://github.com/ARMmbed/simple-mbed-cloud-client
     ```
-    If you do not have an Mbed OS project to add, please create one with 'mbed new' and then 'mbed add' with above step.
+
+    If you do not have an Mbed OS project to add, you can create one with `mbed new <your_application_name>` and then the `mbed add` step above.
+
+2. Configure the API key for your Pelion Portal account.
+
+     If you don't have an API key available, then login in [Pelion IoT Platform portal](https://portal.mbedcloud.com/), navigate to 'Access Management', 'API keys' and create a new one. Then specify the API key as global `mbed` configuration:
+
     ```
-    $ mbed new <your_test_project_name>
+    $ mbed config -G CLOUD_SDK_API_KEY <your-api-key>
     ```
 
-2. Add your Pelion developer certificate to your project (`mbed_cloud_dev_credentials.c` file).
+3. Install the device management certificate:
 
-3. Reference the library from your main.cpp file, add network and storage drivers; finally initialize the Simple Pelion DM Client library. The is the architecture of a generic Simple Pelion DM Client application:
+    ```
+    $ mbed dm init -d "company.com" --model-name "product-model" -q --force
+    ```
```

Comment:
From the user (developer) point of view, I don't think we should be recommending `-q --force`